### PR TITLE
Correctly disable previous year editors

### LINKED ISSUE #137: The type and category in the proposal editor are not filled in when you open proposal editor for previous year
Because the editor contains fields for the current year rather than the year the workshop was from.

These editors should be locked down when looking at previous years anyway...

### PR DESCRIPTION
**Pull request by krzys-h**
* ~~Fix category/type fields for workshops from previous years (closes #137)~~ - see #144 since it targets `master`
* All text fields in editors that are not editable are disabled (rather than throwing an error on save attempt)
* It is now impossible to edit qualification results for previous year workshops
* The buttons to apply for workshops are hidden when looking at previous year program page

### CODE REVIEW

**Comment by cytadela8** [comment]
Open this PR ageist future branch.

**Comment by krzys-h** [comment]
... you could have just clicked the edit button yourself :P

**Comment by cytadela8** [comment]
Ohh, I though that's one thing you can't change. Sorry!

**Comment by krzys-h** [comment]
It used to be like that, but you can change it since like... a few years? :P

**Comment by cytadela8** [comment]
I'm getting old :rofl:

File: wwwapp/models.py [hunk 1]
```diff
@@ -295,9 +295,13 @@ class Workshop(models.Model):
...
     max_points = models.DecimalField(null=True, blank=True, decimal_places=1, max_digits=5)
 
+    def is_editable(self):
+        return (not hasattr(self, 'type') or self.type.year == settings.CURRENT_YEAR) and \
+               datetime.datetime.now().date() < settings.WORKSHOPS_START_DATE
+
     def clean(self):
         super(Workshop, self).clean()
-        if self.type.year != settings.CURRENT_YEAR:
+        if not self.is_editable():
```
Comment (cytadela8) [inline wwwapp/models.py]: 1. Can I get here an override for superusers? (I don't like when system says I can't change something and there is no override)
2. What if somebody wants to edit an workshop after workshops have stared? Cases I can think of: mark workshops as cancelled, add some links or data for participants.

Comment (krzys-h) [reply wwwapp/models.py]: 1. Yeah sure, I actually had to modify settings.CURRENT_YEAR while testing this :joy: A warning saying "you are editing past workshops" would be nice though. We will also have to modify the existing save() method in the model as that blocks changes even through the Django admin page, not to mention normal forms.
2. That's actually a good point. I guess it should allow changes until CURRENT_YEAR is changed.

Comment (krzys-h) [reply wwwapp/models.py]: 1. Actually this is a bigger problem, e.g. you can't edit qualification results as an admin either. We could fix that as well while working on it.

**Comment by krzys-h** [comment]
1. For now, I made it so you can edit the past workshops in Django admin. Support for editing via the site itself will be implemented later as part of #27, #66
2. Done - qualification results will still lock when the workshops start, but everything else can be edited until CURRENT_YEAR changes

**Pull request merged**